### PR TITLE
Fix sundials 6 tests, part 1

### PR DESCRIPTION
**Pull request by drwells**
I presently get a bunch of test failures (listed in #13703) with SUNDIALS 6. A lot of our code looks like

https://github.com/dealii/dealii/blob/8d72163873b452da62abdf59699cd1a775e0e8ba/source/sundials/sunlinsol_wrapper.cc#L243-L257

i.e., we unconditionally abort in SUNDIALS 6. This PR cleans up some SUNDIALS 6 preprocessor checks and fixes the nvector tests.

### CODE REVIEW

**Comment by marcfehling** [comment]
I like the change to GTE!

---

The documentation of make_nvector_view from #13438 looks odd:
https://www.dealii.org/developer/doxygen/deal.II/classSUNDIALS_1_1internal_1_1NVectorView.html#a9578c5b79cf7fbb061e5a2fd49c27f0c

I would probably use the conditional logic around the documentation, rather than moving it into the documentation:
```
#ifdef DEAL_II_SUNDIALS_VERSION_GTE(6,0,0)
  /**
   * ...
   * @note Only available with sundials 6.0 and younger.
   */
   make_nvector_view(..., ...)
#else
  /**
   * ...
   * @note Only available with sundials versions older than 6.0.
   */
  make_nvector_view(...)
#endif
```

This would probably make the `n_vector` test look nicer, too. What do you think?

**Comment by drwells** [comment]
I prefer not having preprocessor stuff in the middle of statements, so yes - lets change that now.

**Comment by drwells** [comment]
It isn't pretty, but getting rid of the preprocessor statements in the tests is worse than the current version - we'd have to write every function call twice. I think we should keep it as-is.

The documentation generated by doxygen (on `master`) is pretty funny:
doxygen-fun

**Review by marcfehling — APPROVED** [review]

File: tests/sundials/n_vector.cc [hunk 1]
```diff
@@ -33,6 +33,12 @@
 
 using namespace SUNDIALS::internal;
 
+// For better compatibility with older versions of SUNDIALS just define this as
+// a static member so we can use it whenever we need it
+#if DEAL_II_SUNDIALS_VERSION_GTE(6, 0, 0)
+static SUNContext nvector_context;
```
Comment (bangerth) [inline tests/sundials/n_vector.cc:39]: Could you call this `global_nvector_context` to make its intent clear? Several of the SUNDIALS solvers have their own local context objects already, so the name would differentiate these more clearly.

**Review by bangerth — COMMENTED** [review]
This looks reasonable if you rename the one variable as requested (if you wouldn't mind).

There are calls to `make_nvector_view()` from within the library itself as well that we will have to figure out what to do with.

File: tests/sundials/n_vector.cc (hunk 1 above)
Comment (drwells) [reply tests/sundials/n_vector.cc]: Sure.

**Review by marcfehling — APPROVED** [review]

**Review by kronbichler — APPROVED** [review]

**Pull request merged**